python/runfiles/runfiles.py

```diff
@@ -15,12 +15,126 @@
 """Runfiles lookup library for Bazel-built Python binaries and tests.
 
 See @rules_python//python/runfiles/README.md for usage instructions.
+
+:::{versionadded} VERSION_NEXT_FEATURE
+Support for Bazel's `--incompatible_compact_repo_mapping_manifest` flag was added.
+This enables prefix-based repository mappings to reduce memory usage for large
+dependency graphs under bzlmod.
+:::
 """
+import collections.abc
+from collections import defaultdict
 import inspect
 import os
 import posixpath
 import sys
-from typing import Dict, Optional, Tuple, Union
+from typing import Dict, Iterator, Optional, Tuple, Union
```

Comment:
Please run our pre-commit command on this to format it. There's a lot of trailing whitespace lines (which, annoyingly, github UI doesn't show)